### ISSUE TITLE
Tag a new version for PHP-CS-Fixer 3.0

### ISSUE THREAD

**Issue by ghost** [matt-allan/laravel-code-style#54]
Could a new version be tagged so I can upgrade my projects? 

```
  Problem 1
    - Installation request for matt-allan/laravel-code-style ^0.6.0 -> satisfiable by matt-allan/laravel-code-style[0.6.0].
    - Can only install one of: friendsofphp/php-cs-fixer[3.0.x-dev, 2.19.x-dev].
    - Can only install one of: friendsofphp/php-cs-fixer[v3.0.0, 2.19.x-dev].
    - Can only install one of: friendsofphp/php-cs-fixer[v3.0.0-beta.1, 2.19.x-dev].
    - Can only install one of: friendsofphp/php-cs-fixer[v3.0.0-beta.2, 2.19.x-dev].
    - Can only install one of: friendsofphp/php-cs-fixer[v3.0.0-rc.1, 2.19.x-dev].
    - Conclusion: install friendsofphp/php-cs-fixer 2.19.x-dev
    - Installation request for friendsofphp/php-cs-fixer ^3.0 -> satisfiable by friendsofphp/php-cs-fixer[3.0.x-dev, v3.0.0, v3.0.0-beta.1, v3.0.0-beta.2, v3.0.0-rc.1].

```

**Comment by matt-allan** [comment]
Hey,

I would like to add official PHP 8 support before tagging a new version but I haven't had a chance to do it myself. In the interim you can use dev-main anchored to a commit hash.

**Comment by JoshMoreno** [comment]
@matt-allan Thanks a lot for making and maintaining this! I'd like to help out if I can. What's needed for php 8 support?

**Comment by matt-allan** [comment]
@JoshMoreno hey, there's some info here: https://github.com/matt-allan/laravel-code-style/issues/39

I would try updating the test matrix and see what fails. If there aren't any issues we should be good.

**Comment by treyssatvincent** [comment]
FYI I use it with PHP 8 within [FooTeam/php-style](https://gitlab.com/FooTeam/php-style) and I didn't had any issues (of course this don't remove the need of tests :sweat_smile:).

**Comment by owenvoke** [comment]
I'm also testing this with PHP 8, we're currently testing in CI against PHP 8 since https://github.com/matt-allan/laravel-code-style/commit/810d5149f5d26fb28322c83c8987c4f0f26affa0. I don't think there are any issues I've found since then.

**Comment by Temepest74** [comment]
+1. I really hope for an update :)))

**Comment by matt-allan** [comment]
Tests are running on PHP 8 now thanks to #48. However I tried un-skipping the tests and a bunch of them are failing now for some reason. We need to figure out what's wrong with the formatting and either fix it or update the tests.

See #55. I don't have much time to work on this so PRs are welcome.

**Comment by chadyred** [comment]
Why not create a travis CI which create automatically version based on Conventional Commit ? In other word, we will be able to do somethings else instead of lost time in this kind of issue ? I write this because you have enough stars to do that :)

**Comment by matt-allan** [comment]
I’m not going to tag a broken build. As I said above, PRs are welcome.

**Comment by Yahav** [comment]
@matt-allan 
Limiting us to unsupported version of PHPCSFIXER doesn't make any sense IMHO and essentially renders this package unusable. 
Perhaps PHP 8 support shouldn't be considered as the same as upgrading to PHPCSFIXER 3.x.
Tag a PHPCSFIXER 3.x version and then continue working on PHP 8 support.

**Comment by matt-allan** [comment]
>  Limiting us to unsupported version of PHPCSFIXER doesn't make any sense IMHO and essentially renders this package unusable.

You're free to pull dev-main as explained above if you're OK with the risk of running untested code. You're also free to make a PR to get the test suite passing again, at which point I will tag the new release.

>  Perhaps PHP 8 support shouldn't be considered as the same as upgrading to PHP-CS-Fixer.

They're effectively the same thing. With PHP-CS-Fixer 3.x it becomes possible to run this package with PHP 8. While the pre 3.x PHP-CS-Fixer package allowed installation on PHP < 8.0 it did not allow you to run it so it wasn't really an issue. I would like to get the tests running on all PHP versions the package is usable with before tagging a stable release.

To be clear the tests are failing on PHP 7.x too; they were just skipped temporarily because of a bug in upstream.

**Comment by matt-allan** [comment]
I fixed the tests and tagged 0.7.0.
